src/workerd/jsg/modules.c++

```diff
@@ -429,4 +451,108 @@ v8::Local<v8::WasmModuleObject> compileWasmModule(jsg::Lock& js,
       v8::MemorySpan<const uint8_t>(code.begin(), code.size())));
 }
 
+// ======================================================================================
+// Node.js module types
+
+jsg::Ref<NodeJsModuleContext> ModuleRegistry::NodeJsModuleInfo::initModuleContext(
+    jsg::Lock& js,
+    kj::StringPtr name) {
+  return jsg::alloc<NodeJsModuleContext>(js.v8Isolate, kj::Path::parse(name));
+}
+
+v8::Local<v8::Value> NodeJsModuleContext::require(kj::String specifier, v8::Isolate* isolate) {
+  static const std::set<kj::StringPtr> NODEJS_BUILTINS {
```

Comment:
Adding to this list is a semver-major change in Node.js and is something we'd probably need to treat as a breaking change (and thus required a compat flag for as we move forward)